### PR TITLE
perf: improve some instances in LinearMap

### PR DESCRIPTION
**Pull request by ChrisHughes24**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by ChrisHughes24** [comment]
!bench

File: Mathlib/Algebra/Module/LinearMap.lean [hunk 1]
```diff
@@ -873,7 +872,7 @@ variable [Module R₁ N₁] [Module R₂ N₂] [Module R₃ N₃]
 variable {σ₁₂ : R₁ →+* R₂} {σ₂₃ : R₂ →+* R₃} {σ₁₃ : R₁ →+* R₃} [RingHomCompTriple σ₁₂ σ₂₃ σ₁₃]
 
 /-- The constant 0 map is linear. -/
-instance : Zero (M →ₛₗ[σ₁₂] M₂) :=
+protected instance zero : Zero (M →ₛₗ[σ₁₂] M₂) :=
```
Comment (eric-wieser) [inline Mathlib/Algebra/Module/LinearMap.lean:875]:
```suggestion
protected instance instZero : Zero (M →ₛₗ[σ₁₂] M₂) :=
```
And throughout

**Comment by ChrisHughes24** [comment]
!bench

**Comment by ChrisHughes24** [comment]
[Speed diff](http://speed.lean-fro.org/mathlib4/compare/b7947944-dc17-42cf-916f-7f90f1dda49d/to/e7b27246-a3e6-496a-b552-ff4b45c7236e?hash2=38c3501c5e5d6c16f1c632e1428f09207e19dd49)  for the last commit (changing the module instance)

**Comment by ChrisHughes24** [comment]
!bench

**Comment by digama0** [comment]
Note: I have pushed an update to the lean toolchain because this PR was on a [buggy version](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Potential.20data.20loss.20from.20.60lake.20clean.60.20with.204.2E2.2E0-rc2.2F3/near/397875701) of the toolchain. **WARNING**: checking out old commits of this PR using v4.2.0-rc2 or v4.2.0-rc3 can cause `lake clean` to delete your mathlib folder! If you need to do so, make sure to delete `lakefile.olean` manually before running any `lake` commands.

**Pull request closed without merging**